lib/linter.js

```diff
@@ -564,14 +566,18 @@ function parse(text, providedParserOptions, parserName, filePath) {
             return {
                 success: true,
                 ast: parseResult.ast,
-                services: parseResult.services || {}
+                services: parseResult.services || {},
+                scopeManager: parseResult.scopeManager,
+                visitorKeys: parseResult.visitorKeys
```

Comment:
Now that `parse` doesn't need to return `parserServices` separately, maybe it would be better for `parse` to just return a `SourceCode` instance.

```js
return {
    success: true,
    sourceCode: new SourceCode({
        text,
        ast: parseResult.ast,
        services: parseResult.services || {},
        scopeManager: parseResult.scopeManager,
        visitorKeys: parseResult.visitorKeys
    })
};
```